README.md

```diff
@@ -23,72 +23,34 @@
 This library fills the need for a simple way for **library developers** to be able to observe the lifecycle of an Angular 
 component.
 
-Unfortunately this is non-trivial with simple Angular due to how ngcc compiles components, so this library hooks into 
-some of the newer private API's of Ivy. As such, the future compatibility of this library is not guaranteed, however it 
-is tested with the version of Angular listed above. Hopefully some day Angular will build the tools to either obsolete 
-this library entirely, or at least let it hook into stable APIs so breakages are limited to Angular's breaking releases.    
-
-
 ## Example
 
 Let's say we're building a simple library function that automatically unsubscribes from observables that were manually 
 subscribed to within a component. We'll implement this as an RxJS operator that can be used as follows:
 
 ```ts
-// ./src/app/lib-example/lib-example.component.ts#L13-L13
+// ./src/app/lib-example/lib-example.component.ts#L12-L12
 
 public timer$ = interval(500).pipe(automaticUnsubscribe(this));
 ````
 
-To build this lib, first we need to create a decorator that the implementing developers can add to their component.
-
+In order to create this operator, we can do the following:
 ```ts
-// ./src/app/lib-example/lib-example.ts#L1-L14
+// ./src/app/lib-example/lib-example.ts#L1-L8
 
-import { decorateObservableLifecycle, getLifecycleHooks } from 'ngx-observable-lifecycle';
+import { getObservableLifecycle } from 'ngx-observable-lifecycle';
 import { Observable } from 'rxjs';
 import { takeUntil } from 'rxjs/operators';
 
-export function AutomaticUnsubscribe(): ClassDecorator {
-  return function (target) {
-    decorateObservableLifecycle(target, {
-      hooks: {
-        onDestroy: true,
-      },
-      incompatibleComponentError: new Error(`You must use @AutomaticUnsubscribe with a directive or component.`),
-    });
-  };
+export function automaticUnsubscribe<T>(component): (source: Observable<T>) => Observable<T> {
```

Comment:
[minor] `automaticUnsubscribe` really sounds like a decorator that'd handle this for you.

I think calling the operator `takeUntilDestroyed` would make more sense